### PR TITLE
Finished bonus step and blue requirements

### PR DESCRIPTION
**Pull request by isomoth**
The project is good enough given the big time constrains this week. The things I struggled with were mainly due to lack of attention to detail: 

- A typo on the "accessories" category
- Not realizing that the user needs to click on an option in the drop-down menu in order to initiate the game
- Misplacing the generateBoard( ) function

A big shout out to my teammates, who with their fresh pair of eyes helped me find these small details.

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -218,108 +220,146 @@ const generateBoard = () => {
...
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  const { category, value } = currentQuestion;
+  // Comparison for key pairs with string values
+  if (category === "hair" || category === "eyes") {
+    if (secret[category] === value) {
+      filterCharacters(true);
+    } else {
+      filterCharacters();
```
Comment (brunsant) [inline code/script.js:267]: In one of our group conversations, people were discussing if we should write "false" or leave the boolean empty. I found an article that really helped me understand this: 
https://medium.com/swlh/avoid-null-booleans-in-java-4a5cd9b23bca

File: code/script.js [hunk 2]
```diff
@@ -218,108 +220,146 @@ const generateBoard = () => {
...
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  // Send the user to the Win Or Lose screen
+  winOrLoseSection.style.display = "block";
+  // Determine who won by comparing the user's guess with the secret character
+  if (personToCheck === secret.name) {
+    winOrLoseMessage.innerHTML = "You win!";
+  } else {
+    winOrLoseMessage.innerHTML = "Nope sorry, I win!";
```
Comment (brunsant) [inline code/script.js:354]: Very nice having the message using innerHTML!

**Review by brunsant — APPROVED** [review]
Amazing job with this project!
The game works perfectly, filtering the characters depending on the parameters with a nice win or lose section in the end. Really liked the color palette!!!

The code itself was very clean and easy to follow, with no mistakes in it.  The comments you made are concise and really helpful to understand the info. I'm really impressed with your work!
It was a pleasure to review your code!

File: code/script.js [hunk 3]
```diff
@@ -218,108 +220,146 @@ const generateBoard = () => {
...
-  // What else should happen when we start the game?
-}
+  // Set charactersInPlay array to include all the characters we start with
+  charactersInPlay = CHARACTERS;
+  // Invoke the function that draws the game board
+  generateBoard();
+  // Pick a secret character
+  setSecret();
+  // Hide the win or lose screen
+  winOrLoseSection.style.display = "none";
```
[the remark this answers is not in the document]
Comment (isomoth) [reply code/script.js:243]: Oh right, that could be another way to do it! It's nice to find out about other ways to do things :)

File: code/script.js (hunk 1 above)
Comment (isomoth) [reply code/script.js:267]: Oh wow, that's a really useful article! I myself wasn't so sure about it, but now it's clear to me. Thanks a lot Bruna!

**Pull request closed without merging**